vmm/src/config.rs

```diff
@@ -3690,6 +3690,219 @@ mod tests {
+
+        let mut invalid_config = valid_config.clone();
+        invalid_config.net_fds = Some(vec![RestoredNetConfig {
+            id: "netx".to_string(),
+            num_fds: 4,
+            fds: Some(vec![3, 4, 5, 6]),
+        }]);
+        assert_eq!(
+            invalid_config.validate(&snapshot_vm_config),
+            Err(ValidationError::InvalidIdentifier("netx".to_string()))
+        );
+
+        invalid_config.net_fds = Some(vec![
+            RestoredNetConfig {
+                id: "net0".to_string(),
+                num_fds: 4,
+                fds: Some(vec![3, 4, 5, 6]),
+            },
+            RestoredNetConfig {
+                id: "net0".to_string(),
+                num_fds: 4,
+                fds: Some(vec![3, 4, 5, 6]),
+            },
+        ]);
+        assert_eq!(
+            invalid_config.validate(&snapshot_vm_config),
+            Err(ValidationError::IdentifierNotUnique("net0".to_string()))
+        );
+
+        invalid_config.net_fds = Some(vec![RestoredNetConfig {
+            id: "net0".to_string(),
+            num_fds: 4,
+            fds: Some(vec![3, 4, 5, 6]),
+        }]);
+        assert_eq!(
+            invalid_config.validate(&snapshot_vm_config),
+            Err(ValidationError::RestoreMissingRequiredNetId(
+                "net1".to_string()
+            ))
+        );
+
+        invalid_config.net_fds = Some(vec![
+            RestoredNetConfig {
+                id: "net0".to_string(),
+                num_fds: 4,
+                fds: Some(vec![3, 4, 5, 6]),
+            },
+            RestoredNetConfig {
+                id: "net1".to_string(),
+                num_fds: 2,
+                fds: Some(vec![7, 8]),
+            },
+            RestoredNetConfig {
+                id: "net2".to_string(),
+                num_fds: 2,
+                fds: Some(vec![9, 10]),
+            },
+        ]);
+        assert_eq!(
+            invalid_config.validate(&snapshot_vm_config),
+            Err(ValidationError::RestoreNonFdNetIdNotExpected(
+                "net2".to_string()
+            ))
+        );
+
+        invalid_config.net_fds = Some(vec![RestoredNetConfig {
+            id: "net0".to_string(),
+            num_fds: 2,
+            fds: Some(vec![3, 4]),
+        }]);
+        assert_eq!(
+            invalid_config.validate(&snapshot_vm_config),
+            Err(ValidationError::RestoreNetFdCountMismatch(
+                "net0".to_string(),
+                2,
+                4
+            ))
+        );
+
+        invalid_config.net_fds = Some(vec![RestoredNetConfig {
+            id: "net0".to_string(),
+            num_fds: 4,
+            fds: Some(vec![3, 4]),
+        }]);
+        assert_eq!(
+            invalid_config.validate(&snapshot_vm_config),
+            Err(ValidationError::RestoreNetFdCountMismatch(
+                "net0".to_string(),
+                2,
+                4
+            ))
+        );
```

Comment:
Such scenario should be a programming error (e.g. something that user can not control), and hence should be caught as an assertion, rather than returning as an error to user.